traitsui/testing/tester/examples/demo_examples.py

```diff
@@ -0,0 +1,53 @@
+""" The examples in this file showcase UI Tester being used on some of the
+already existing demos for TraitsUI itself.
+"""
+import os
+import time
+
+import pkg_resources
+
+from traitsui.testing.tester import command, locator, query
+from traitsui.testing.tester.ui_tester import UITester
+
+DEMO = pkg_resources.resource_filename("traitsui", "examples/demo")
+
+
+def load_demo(file_path, variable_name="demo"):
+    with open(file_path, "r", encoding="utf-8") as f:
+        content = f.read()
+    globals_ = globals().copy()
+    exec(content, globals_)
+    return globals_[variable_name]
+
+
+if __name__ == '__main__':
+    # Test converter.py in examples/demo/Applications
+    filepath = os.path.join(
+        DEMO, "Applications", "converter.py"
+    )
+    demo = load_demo(filepath, "popup")
+    tester = UITester(delay=1500)
+    with tester.create_ui(demo) as ui:
+        input_amount = tester.find_by_name(ui, "input_amount")
+        input_amount.delay = 200
+        output_amount = tester.find_by_name(ui, "output_amount")
+        input_units = tester.find_by_name(ui, "input_units")
+        output_units = tester.find_by_name(ui, "output_units")
+
+        for _ in range(4):
+            input_amount.perform(command.KeyClick("Backspace"))
+        input_amount.perform(command.KeySequence("14.0"))
+        assert output_amount.inspect(query.DisplayedText())[:4] == "1.16"
+
+        tester.find_by_id(ui, "Undo").perform(command.MouseClick())
+
+        assert output_amount.inspect(query.DisplayedText()) == "1.0"
+
+        for _ in range(3):
+            input_amount.perform(command.KeyClick("Backspace"))
+        input_units.locate(locator.Index(5)).perform(command.MouseClick())
+        output_units.locate(locator.Index(3)).perform(command.MouseClick())
+
+        assert output_amount.inspect(query.DisplayedText())[:3] == "1.6"
+
+        time.sleep(2)
```

Comment:
this is also implemented in https://github.com/enthought/traitsui/blob/master/integrationtests/test_all_examples.py, but it feels appropriate here as a good example of using the tester.  Should this code be removed from test_all_examples, or removed here, or neither?

Reply:
In fact, many of the other examples here for UI Tester could be effectively replaced by simply loading the corresponding TraitsUI demo and utilizing the tester on it like is done here.  The examples directly borrow the models/HasTraits classes etc. from the TraitsIO examples

Reply:
@kitchoi what are your thoughts about pulling this load_demo function into a module, and then for each example of the UI Tester, we could just load the corresponding TraitsUI demo for the editor, and interact with it via the tester, to show how the tester can be used on each editor.  I think this would be cleaner that how it is currently being done, and it is also seems  nice that the examples are consistent.

Reply:
Ah, I started doing this, but there are actually a few annoyances.  For example, in https://github.com/enthought/traitsui/blob/master/traitsui/examples/demo/Standard_Editors/CheckListEditor_simple_demo.py the readonly and custom style editors are both named 'checklist' so using `find_by_name` with the tester does not work.  It seems to make some of the examples more tester friendly it might require adjusting the already existing examples in TraitsUI.  In the most recent commit I made this change to the Button Editor example just for reference as to how this change would look.  If we decide against loading the traitsui examples I can simply revert this commit. 
Going to pause working on this for now until I confirm the best way to proceed.